scripts/install_quantization_libs.py

```diff
@@ -35,29 +18,43 @@ def clone_or_pull_repo(repo_url, repo_location_path):
         subprocess.run(f"git clone {repo_url} {repo_location_path}", shell=True, check=True)
 
 
+def process_setup_file_for_autoawq(setup_file_path):
+    print("Processing setup.py for AutoAWQ")
+
+    with open(setup_file_path, "r") as file:
+        setup_content = file.read()
+
+    # Use regex to match any line that starts with IS_CPU_ONLY = and modify it to IS_CPU_ONLY = False
+    setup_content = re.sub(r"(IS_CPU_ONLY\s*=\s*.*)", r"\1\nIS_CPU_ONLY = False", setup_content)
+
+    # Write the modified content back to setup.py
+    with open(setup_file_path, "w") as file:
+        file.write(setup_content)
+
+
 def install_autoawq_from_source():
     """Install the AutoAWQ and AutoAWQ_kernels packages from GitHub."""
     print("Installing AutoAWQ and AutoAWQ_kernels packages.")
-
     autoawq_repo_name = "AutoAWQ"
     autoawq_kernels_repo_name = "AutoAWQ_kernels"
 
     autoawq_repo_path = os.path.join(EXTERNAL_REPOS_DIR, autoawq_repo_name)
     kernels_repo_path = os.path.join(EXTERNAL_REPOS_DIR, autoawq_kernels_repo_name)
 
+    print("Installing AutoAWQ_kernels package.")
     clone_or_pull_repo(f"https://github.com/casper-hansen/{autoawq_kernels_repo_name}", kernels_repo_path)
-    kernels_setup_file_path = os.path.join(kernels_repo_path, "setup.py")
-    process_setup_file(kernels_setup_file_path)
     subprocess.run(
         f"cd {kernels_repo_path} && {sys.executable} -m pip install .",
         shell=True,
         check=True,
         env=os.environ,
     )
 
+    print("Installing AutoAWQ package.")
     clone_or_pull_repo(f"https://github.com/casper-hansen/{autoawq_repo_name}", autoawq_repo_path)
-    autoawq_setup_file_path = os.path.join(autoawq_repo_path, "setup.py")
-    process_setup_file(autoawq_setup_file_path)
+    if os.environ.get("IMAGE_FLAVOR") in ["cuda", "rocm"]:
+        autoawq_setup_file_path = os.path.join(autoawq_repo_path, "setup.py")
+        process_setup_file_for_autoawq(autoawq_setup_file_path)
```

Comment:
we only use these scripts in these two images, what are we checking here ?

Reply:
I just realized i forgot to add cuda-ort. But the idea is to patch the setup.py to remove the installation of ipex for the none cpu based image https://github.com/casper-hansen/AutoAWQ/blob/main/setup.py#L10. Just to confirm we want to have ipex for the cpu image or is better not to have it (as we could have a amd cpu)?

Reply:
it is true it is redundant, i will remove it